v2/stream/new/flow.go

```diff
@@ -0,0 +1,193 @@
+package new
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+var initializeTimeout = 3 * time.Second
+var authRetryDelayMultiplier = time.Second
+var authRetryCount = 15
+
+// initialize performs the initial flow:
+// 1. wait to be welcomed
+// 2. authenticates (and waits for the response)
+// 3. subscribes (and waits for the response)
+//
+// If it runs into retriable issues with authentication it retries for a while
+func (c *client) initialize(ctx context.Context) error {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, initializeTimeout)
+	defer cancel()
+
+	if err := c.readConnected(ctxWithTimeout); err != nil {
+		return err
+	}
+
+	var retryErr error
+	// If we run into a retriable error during auth we should try
+	// again after a delay
+	for i := 0; i < authRetryCount; i++ {
+		if i > 0 {
+			n := i
+			if n > 10 {
+				n = 10
+			}
+			sleepDuration := authRetryDelayMultiplier * time.Duration(n*n)
```

Comment:
The `authRetryDelayMultiplier` name or type is misleading.

Reply:
I am open to suggestions